src/main/webapp/populateData.js

```diff
@@ -47,3 +47,33 @@ function populateGmail() {
       });
 }
 
+/**
+ * Populate Tasks container with user information
+ */
+function populateTasks() {
+  // Get Container for Tasks content
+  const tasksContainer = document.querySelector('#tasks');
+
+  // Get list of tasks from user's Tasks account
+  // and display the task titles from all task lists on the screen
+  fetch('/tasks')
+      .then((response) => {
+        // If response is a 403, user is not authenticated
+        if (response.status === 403) {
+          throw new Error();
```

Comment:
You may want to define a specific type for authentication errors, and check for this type in the catch below. That way, errors unrelated to authentication can be handled differently (or not at all).

See https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Error

Reply:
Fixed! Thanks for the suggestion - I didn't know this was also a Javascript feature